### PR TITLE
[Misc] Add battle music for Legendary Pokémon forms

### PR DESCRIPTION
**Pull request by Madmadness65**
## What are the changes the user will see?

Adds more battle music for Legendary Pokémon with alternate forms, specifically Dialga, Palkia, and Calyrex. Dialga and Palkia will get their Legends Arceus Origin Form battle theme, and Calyrex will get its Ice & Shadow Rider battle theme.

## Why am I making these changes?

These specific Pokémon have battle themes for their alternate forms, and can be utilized if they are either encountered in the wild or as a static boss encounter, like what I mentioned in #3054 .

## What are the changes from a developer perspective?

Enemy Dialga, Palkia, and Calyrex now play different music based on their form key, with the existing music playing for their base forms, and the new tracks playing for their alternate forms.

### Screenshots/Videos

https://github.com/user-attachments/assets/7c64ce5c-9c69-467c-8a0f-d19d1fdf2b04

## How to test the changes?

There isn't a way to force the form index of an opponent at present, so the easiest way to test would be to temporarily add Dialga, Palkia, or Calyrex to the `getSpeciesFormIndex` in `battle-scene.ts` and encounter the forms randomly to see the music changes.

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [ ] If I have text, did I add placeholders for them in locales?
- [x] Have I tested the changes (manually)?
  - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by CodeTappert — APPROVED** [review]
More modern music is always a +!

**Review by DayKev — APPROVED** [review]

File: src/battle.ts [hunk 1]
```diff
@@ -275,7 +275,12 @@ export default class Battle {
             return "battle_legendary_sinnoh";
           }
           if (pokemon.species.speciesId === Species.DIALGA || pokemon.species.speciesId === Species.PALKIA) {
-            return "battle_legendary_dia_pal";
+            if (pokemon.getFormKey() === "") {
+              return "battle_legendary_dia_pal";
+            }
+            if (pokemon.getFormKey() === "origin") {
+              return "battle_legendary_origin_form";
```
Comment (chaosgrimmon) [inline src/battle.ts:282]: also forme?

File: src/battle-scene.ts [hunk 2]
```diff
@@ -1897,6 +1897,8 @@ export default class BattleScene extends SceneBase {
       return 22.770;
     case "battle_legendary_dia_pal": //ORAS Dialga & Palkia Battle
       return 16.009;
+    case "battle_legendary_origin_form": //LA Origin Dialga & Palkia Battle
```
Comment (chaosgrimmon) [inline src/battle-scene.ts:1900]: forme?

File: src/locales/en/bgm-name.json [hunk 3]
```diff
@@ -45,6 +45,7 @@
   "battle_legendary_lake_trio": "ORAS Lake Guardians Battle",
   "battle_legendary_sinnoh": "ORAS Sinnoh Legendary Battle",
   "battle_legendary_dia_pal": "ORAS Dialga & Palkia Battle",
+  "battle_legendary_origin_form": "LA Origin Dialga & Palkia Battle",
```
Comment (chaosgrimmon) [inline src/locales/en/bgm-name.json:48]: forme here too, then

**Review by KimJeongSun — APPROVED** [review]
cool update!

**Review by chaosgrimmon — APPROVED** [review]
Style nit, just to avoid an unnecessary terminology divergence from the official games.

**Comment by chaosgrimmon** [comment]
oh no that automatically merges

**Comment by Madmadness65** [comment]
> Style nit, just to avoid an unnecessary terminology divergence from the official games.

Thanks, I merged it before your review was posted, so I fixed that in d15c01378d1 .

**Pull request merged**